docs/intro/testlib/index.md

```diff
@@ -1,16 +1,16 @@
-如果你正在使用 C++ 出一道算法竞赛题目，Testlib 是编写相关程序（generator, validator, checker, interactor）时的优秀辅助工具。它是俄罗斯和其他一些国家的出题人的必备工具，许多比赛也都在用它：ROI、ICPC 区域赛、所有 Codeforces round……
+If you are using C++ to create an OI problem, Testlib is an excellent helper tool when writing related programs (generator, validator, checker, interactor). It has become indispensable for OI problem authors from Russia and some other countries, and many competitions are using it as well: ROI, ICPC regional competitions, all Codeforces round ...
 
-Testlib 库仅有 `testlib.h` 一个文件，使用时仅需在所编写的程序开头添加 `#include "testlib.h"` 即可。
+The Testlib library has only one file called `testlib.h`, so you only need to add `#include "testlib.h"` at the beginning of the program.
 
-Testlib 的具体用途：
+The specific usage of Testlib:
 
-- 编写 [Generator](./generator.md) ，即数据生成器。
-- 编写 [Validator](./validator.md) ，即数据校验器，判断生成数据是否符合题目要求，如数据范围、格式等。
-- 编写 [Interactor](./interactor.md) ，即交互器，用于交互题。
-- 编写 [Checker](./checker.md) ，即 [Special Judge](../spj.md) 。
+- Write [generator](./generator.md), which is the data generator.
+- Write [validator](./validator.md) to determine whether the generated data meets the requirements of the problem, such as data range, format, etc.
+- Write [interactor](./interactor.md), the interactive device, which is used for interactive questions.
+- Write [checker](./checker.md), which is [Special Judge](../spj.md).
 
-Testlib 与 Codeforces 开发的 [Polygon](https://polygon.codeforces.com/) 出题平台完全兼容。
+Testlib is fully compatible with [Polygon](https://polygon.codeforces.com/) which is developed by Codeforces.
 
- `testlib.h` 在 2005 年移植自 `testlib.pas` ，并一直在更新。Testlib 与绝大多数编译器兼容，如 VC++ 和 GCC g++，并兼容 C++11。
+ `testlib.h` was migrated from `testlib.pas` in 2005 and has been updated ever since. Testlib is compatible with most compilers, such as VC++, GCC g++, and compatible with C++11.
```

Comment:
```suggestion
 `testlib.h` was migrated from `testlib.pas` in 2005 and has been updated ever since. Testlib is compatible with most compilers, such as VC++, GCC g++, and is compatible with C++11.
```